mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/modules/CassandraMessageModule.java

```diff
@@ -89,33 +90,46 @@ public CassandraMessageModule() {
                     .ifNotExists()
                     .addPartitionKey(CassandraMessageIds.MESSAGE_ID, timeuuid())
-                    .addColumn(CassandraMessageTable.INTERNAL_DATE, timestamp())
-                    .addColumn(CassandraMessageTable.BODY_START_OCTET, cint())
-                    .addColumn(CassandraMessageTable.BODY_OCTECTS, bigint())
-                    .addColumn(CassandraMessageTable.TEXTUAL_LINE_COUNT, bigint())
-                    .addColumn(CassandraMessageTable.FULL_CONTENT_OCTETS, bigint())
-                    .addColumn(CassandraMessageTable.BODY_CONTENT, blob())
-                    .addColumn(CassandraMessageTable.HEADER_CONTENT, blob())
-                    .addUDTListColumn(CassandraMessageTable.ATTACHMENTS, SchemaBuilder.frozen(CassandraMessageTable.ATTACHMENTS))
-                    .addUDTListColumn(CassandraMessageTable.PROPERTIES, SchemaBuilder.frozen(CassandraMessageTable.PROPERTIES))));
+                    .addColumn(CassandraMessageV1Table.INTERNAL_DATE, timestamp())
+                    .addColumn(CassandraMessageV1Table.BODY_START_OCTET, cint())
+                    .addColumn(CassandraMessageV1Table.BODY_OCTECTS, bigint())
+                    .addColumn(CassandraMessageV1Table.TEXTUAL_LINE_COUNT, bigint())
+                    .addColumn(CassandraMessageV1Table.FULL_CONTENT_OCTETS, bigint())
+                    .addColumn(CassandraMessageV1Table.BODY_CONTENT, blob())
+                    .addColumn(CassandraMessageV1Table.HEADER_CONTENT, blob())
+                    .addUDTListColumn(CassandraMessageV1Table.ATTACHMENTS, SchemaBuilder.frozen(CassandraMessageV1Table.ATTACHMENTS))
+                    .addUDTListColumn(CassandraMessageV1Table.PROPERTIES, SchemaBuilder.frozen(CassandraMessageV1Table.PROPERTIES))),
+            new CassandraTable(CassandraMessageV2Table.TABLE_NAME,
+                SchemaBuilder.createTable(CassandraMessageV2Table.TABLE_NAME)
+                    .ifNotExists()
+                    .addPartitionKey(CassandraMessageIds.MESSAGE_ID, timeuuid())
+                    .addColumn(CassandraMessageV2Table.INTERNAL_DATE, timestamp())
+                    .addColumn(CassandraMessageV2Table.BODY_START_OCTET, cint())
+                    .addColumn(CassandraMessageV2Table.BODY_OCTECTS, bigint())
+                    .addColumn(CassandraMessageV2Table.TEXTUAL_LINE_COUNT, bigint())
+                    .addColumn(CassandraMessageV2Table.FULL_CONTENT_OCTETS, bigint())
+                    .addColumn(CassandraMessageV2Table.BODY_CONTENT, timeuuid())
+                    .addColumn(CassandraMessageV2Table.HEADER_CONTENT, timeuuid())
+                    .addUDTListColumn(CassandraMessageV2Table.ATTACHMENTS, SchemaBuilder.frozen(CassandraMessageV2Table.ATTACHMENTS))
+                    .addUDTListColumn(CassandraMessageV2Table.PROPERTIES, SchemaBuilder.frozen(CassandraMessageV2Table.PROPERTIES))));
         types = ImmutableList.of(
```

Comment:
I don't agree with you, types are part of the schema and should be versioned.
They are also closed to a table, so why not to keep them in the versioned table definition?
(I our case they should be in CassandraMessageV2Table)